static/js/settings_profile_fields.js

```diff
@@ -166,68 +167,77 @@ function clear_form_data() {
         $field_url_pattern_elem.hide();
         $field_elem.hide();
     }
 }
 
-function read_field_data_from_form(field_type_id, field_elem, old_field_data) {
+function read_field_data_from_form(field_type_id, $profile_field_form, old_field_data) {
     // Only read field data if we are creating a select field
     // or external account field.
     if (field_type_id === field_types.SELECT.id) {
-        return read_select_field_data_from_form(field_elem, old_field_data);
+        return read_select_field_data_from_form($profile_field_form, old_field_data);
     } else if (field_type_id === field_types.EXTERNAL_ACCOUNT.id) {
-        return read_external_account_field_data(field_elem);
+        return read_external_account_field_data($profile_field_form);
     }
     return undefined;
 }
 
-function create_profile_field(e) {
-    e.preventDefault();
-    e.stopPropagation();
+function open_custom_profile_field_form_modal() {
+    const html_body = render_add_new_custom_profile_field_form({
+        realm_default_external_accounts: page_params.realm_default_external_accounts,
+        custom_profile_field_types: page_params.custom_profile_field_types,
+    });
 
-    let field_data = {};
-    const field_type = $("#profile_field_type").val();
-    const opts = {
-        success_continuation: clear_form_data,
-    };
-    field_data = read_field_data_from_form(
-        Number.parseInt(field_type, 10),
-        $(".new-profile-field-form"),
-    );
+    function create_profile_field() {
+        let field_data = {};
+        const field_type = $("#profile_field_type").val();
+        field_data = read_field_data_from_form(
+            Number.parseInt(field_type, 10),
+            $(".new-profile-field-form"),
+        );
+        const data = {
+            name: $("#profile_field_name").val(),
+            hint: $("#profile_field_hint").val(),
+            field_type,
+            field_data: JSON.stringify(field_data),
+        };
+        const url = "/json/realm/profile_fields";
+        dialog_widget.submit_api_request(channel.post, url, data);
+    }
 
-    const form_data = {
-        name: $("#profile_field_name").val(),
-        field_type,
-        hint: $("#profile_field_hint").val(),
-        field_data: JSON.stringify(field_data),
-    };
+    function set_up_form_fields() {
+        set_up_select_field();
+        set_up_external_account_field();
+        clear_form_data();
+    }
 
-    settings_ui.do_settings_change(
-        channel.post,
-        "/json/realm/profile_fields",
-        form_data,
-        $("#admin-add-profile-field-status").expectOne(),
-        opts,
-    );
-    update_profile_fields_table_element();
+    dialog_widget.launch({
+        html_heading: $t_html({defaultMessage: "Add a new custom profile field"}),
+        html_body,
+        on_click: create_profile_field,
+        post_render: set_up_form_fields,
+        loading_spinner: true,
+    });
```

Comment:
You can add a [help link](https://github.com/zulip/zulip/blob/7d8be670a58016ef7258deaee19451e92dd31023/static/js/dialog_widget.js#L81) (https://zulip.com/help/add-custom-profile-fields)

Reply:
Done